lms/djangoapps/courseware/views/views.py

```diff
@@ -923,6 +924,7 @@ def _get_cert_data(student, course, course_key, is_active, enrollment_mode):
     show_generate_cert_btn = (
         is_active and CourseMode.is_eligible_for_certificate(enrollment_mode)
         and certs_api.cert_generation_enabled(course_key)
+        and course_key and get_course_by_id(course_key).may_certify()
```

Comment:
@sanfordstudent FYI, I just moved this unavailable date logic into showing the button or not, rather than generating a different message based on a state returned by the API.